lib/ruby_saml/utils.rb

```diff
@@ -84,77 +87,44 @@ def self.parse_duration(duration, timestamp=Time.now.utc)
       datetime.to_time.utc.to_i + (durHours * 3600) + (durMinutes * 60) + durSeconds
     end
 
-    # Return a properly formatted x509 certificate
-    #
-    # @param cert [String] The original certificate
-    # @return [String] The formatted certificate
+    # Formats one or many X.509 certificate(s) to canonical
+    # PEM format with 64-char lines and BEGIN/END headers.
     #
-    def self.format_cert(cert)
-      # don't try to format an encoded certificate or if is empty or nil
-      if cert.respond_to?(:ascii_only?)
-        return cert if cert.nil? || cert.empty? || !cert.ascii_only?
-      elsif cert.nil? || cert.empty? || cert.match(/\x0d/)
-        return cert
-      end
-
-      if cert.scan(/BEGIN CERTIFICATE/).length > 1
-        formatted_cert = []
-        cert.scan(/-{5}BEGIN CERTIFICATE-{5}[\n\r]?.*?-{5}END CERTIFICATE-{5}[\n\r]?/m) do |c|
-          formatted_cert << format_cert(c)
-        end
-        formatted_cert.join("\n")
-      else
-        cert = cert.gsub(/-{5}\s?(BEGIN|END) CERTIFICATE\s?-{5}/, "")
-        cert = cert.gsub(/\r/, "")
-        cert = cert.gsub(/\n/, "")
-        cert = cert.gsub(/\s/, "")
-        cert = cert.scan(/.{1,64}/)
-        cert = cert.join("\n")
-        "-----BEGIN CERTIFICATE-----\n#{cert}\n-----END CERTIFICATE-----"
-      end
+    # @param cert [String] The original certificate(s)
+    # @param multi [true|false] Whether to return multiple keys delimited by newline
+    # @return [String|nil] The formatted certificate(s)
+    def format_cert(cert, multi: true)
```

Comment:
Here `multi: true` because that was the legacy behavior. It doesn't really make sense tho...